src/Nethermind/Nethermind.Taiko/Rpc/TaikoEngineRpcModule.cs

```diff
@@ -182,7 +182,7 @@ void CommitAndDisposeBatch(Batch batch)
                     {
                         worldState.Restore(snapshot);
 
-                        if (executionResult.Error == TransactionResult.BlockGasLimitExceeded && batch.Transactions.Count is not 0)
+                        if (executionResult == TransactionResult.BlockGasLimitExceeded && batch.Transactions.Count is not 0)
```

Comment:
To have this kind of comparisions I think you need to override Equals in TransactionResult?

Reply:
Nice catch!